lib/ansible/modules/cloud/memset/memset_zone_domain.py

```diff
@@ -0,0 +1,270 @@
+#
+# Copyright (c) 2018, Simon Weald <ansible@simonweald.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_zone_domain
+author: "Simon Weald (@analbeard)"
+version_added: "2.6"
+short_description: Create and delete domains in Memset DNS zones.
+notes:
+  - Zone domains can be thought of as a collection of domains, all of which share the
+    same DNS records (i.e. they point to the same IP). An API key generated via the
+    Memset customer control panel is needed with the following minimum scope -
+    I(dns.zone_domain_create), I(dns.zone_domain_delete), I(dns.zone_domain_list).
+  - Currently this module can only create one domain at a time. Multiple domains should
+    be created using C(with_items).
+description:
+    - Manage DNS zone domains in a Memset account.
+options:
+    state:
+        default: present
+        description:
+            - Indicates desired state of resource.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        description:
+            - The API key obtained from the Memset control panel.
+    domain:
+        required: true
+        description:
+            - The zone domain name. Ensure this value has at most 250 characters.
+        aliases: ['name']
+    zone:
+        required: true
+        description:
+            - The zone to add the domain to (this must already exist).
+'''
+
+EXAMPLES = '''
+# Create the zone domain 'test.com'
+- name: create zone domain
+  memset_zone_domain:
+    domain: test.com
+    zone: testzone
+    state: present
+    api_key: 5eb86c9196ab03919abcf03857163741
+  delegate_to: localhost
+'''
+
+RETURN = '''
+memset_api:
+  description: Domain info from the Memset API
+  returned: when changed or state == present
+  type: complex
+  contains:
+    domain:
+      description: Domain name
+      returned: always
+      type: string
+      sample: "example.com"
+    id:
+      description: Domain ID
+      returned: always
+      type: string
+      sample: "b0bb1ce851aeea6feeb2dc32fe83bf9c"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.memset import get_zone_id
+from ansible.module_utils.memset import check_zone_domain
+from ansible.module_utils.memset import memset_api_call
+
+
+def api_validation(args=None):
+    '''
+    Perform some validation which will be enforced by Memset's API (see:
+    https://www.memset.com/apidocs/methods_dns.html#dns.zone_domain_create)
+    '''
+    # zone domain length must be less than 250 chars
+    if len(args['domain']) > 250:
+        stderr = 'Zone domain must be less than 250 characters in length.'
+        module.fail_json(failed=True, msg=stderr, stderr=stderr)
```

Comment:
Just curious, why do you want the error message twice? Anyway msg would be printed.

Reply:
Good point, that was on my snagging list to resolve. I've just pushed a change to fix this and another occurrence.